applications/PoromechanicsApplication/python_scripts/poromechanics_U_Pw_solver.py

```diff
@@ -156,129 +130,224 @@ def AddDofs(self):
 
         # Get the convergence criterion
-        convergence_criterion = self._ConstructConvergenceCriterion(self.settings["convergence_criterion"].GetString())
+        self.convergence_criterion = self._ConstructConvergenceCriterion(self.settings["convergence_criterion"].GetString())
 
         # Solver creation
-        self.Solver = self._ConstructSolver(builder_and_solver,
-                                            scheme,
-                                            convergence_criterion,
+        self.solver = self._ConstructSolver(builder_and_solver,
                                             self.settings["strategy_type"].GetString())
 
         # Set echo_level
-        self.Solver.SetEchoLevel(self.settings["echo_level"].GetInt())
+        self.SetEchoLevel(self.settings["echo_level"].GetInt())
+
+        # Initialize Strategy
+        if self.settings["clear_storage"].GetBool():
+            self.Clear()
+
+        self.solver.Initialize()
 
         # Check if everything is assigned correctly
-        self.Solver.Check()
+        self.Check()
 
-        print ("Initialization UPwSolver finished")
+        KratosMultiphysics.Logger.PrintInfo("UPwSolver", "Solver initialization finished.")
 
     def GetComputingModelPart(self):
         return self.main_model_part.GetSubModelPart(self.computing_model_part_name)
 
-    def GetOutputVariables(self):
-        pass
-
     def ComputeDeltaTime(self):
-        pass
+        return self.main_model_part.ProcessInfo[KratosMultiphysics.DELTA_TIME]
 
-    def SaveRestart(self):
-        pass #one should write the restart file here
+    def Clear(self):
+        self.solver.Clear()
 
-    def Solve(self):
-        if self.settings["clear_storage"].GetBool():
-            self.Clear()
+    def Check(self):
+        self.solver.Check()
 
-        self.Solver.Solve()
+    def SetEchoLevel(self, level):
+        self.solver.SetEchoLevel(level)
 
-    # solve :: sequencial calls
+    def AdvanceInTime(self, current_time):
+        dt = self.ComputeDeltaTime()
+        new_time = current_time + dt
 
-    def InitializeStrategy(self):
-        if self.settings["clear_storage"].GetBool():
-            self.Clear()
+        self.main_model_part.CloneTimeStep(new_time)
+        self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
 
-        self.Solver.Initialize()
+        return new_time
 
     def InitializeSolutionStep(self):
-        self.Solver.InitializeSolutionStep()
+        self.solver.InitializeSolutionStep()
 
     def Predict(self):
-        self.Solver.Predict()
+        self.solver.Predict()
 
     def SolveSolutionStep(self):
-        self.Solver.SolveSolutionStep()
+        is_converged = self.solver.SolveSolutionStep()
+        return is_converged
 
     def FinalizeSolutionStep(self):
-        self.Solver.FinalizeSolutionStep()
+        self.solver.FinalizeSolutionStep()
 
-    # solve :: sequencial calls
+    def Solve(self):
```

Comment:
This should not be called according to riccardo

The FluidSolver issues a warning here

Reply:
Thanks! I also added a warning.